sdk/core/azure-core/inc/azure/core/http/body_stream.hpp

```diff
@@ -211,34 +207,84 @@ namespace Azure { namespace Core { namespace Http {
     int64_t OnRead(Azure::Core::Context const& context, uint8_t* buffer, int64_t count) override;
 
   public:
-#if defined(AZ_PLATFORM_POSIX)
     /**
      * @brief Construct from a file.
      *
-     * @param fd File descriptor.
-     * @param offset Offset in the file to start providing the data from.
-     * @param length Length of the data, in bytes, to provide.
+     * @param file A pointer to an already opened file object that can be used to identify the file.
+     *
+     * @remark The caller owns the file object and needs to open it along with keeping it alive for
+     * the necessary duration. The caller is also responsible for closing it once they are done.
      */
-    FileBodyStream(int fd, int64_t offset, int64_t length)
-        : m_fd(fd), m_baseOffset(offset), m_length(length), m_offset(0)
+    FileBodyStream(FILE* file) : m_hFile(file), m_offset(0)
     {
+      if (file == NULL)
+      {
+        throw std::invalid_argument(
+            "The file object cannot be null and must have been successfully opened.");
+      }
+
+      // Get the current file position, to reset it back, after seeking to the end.
+      if (fgetpos(file, &m_baseOffset))
+      {
+        throw std::runtime_error("Failed to get the file object position.");
+      }
+
+      if (fseek(file, 0, SEEK_END))
+      {
+        throw std::runtime_error("Failed to seek to the end of the file.");
+      }
```

Comment:
why? Are you trying to prevent a future error while reading?  
It would be still possible,  someone might delete the file after this seek or something. Maybe we don't need this

Reply:
> Are you trying to prevent a future error while reading?

No, this is necessary to calculate the file size on behalf of the user, if they don't know it ahead of time.